cli/internal/turbostate/turbostate.go

```diff
@@ -97,45 +88,18 @@ type ParsedArgsFromRust struct {
 	Command            Command `json:"command"`
 }
 
-// GetColor returns the value of the `color` flag.
-func (a ParsedArgsFromRust) GetColor() bool {
-	return a.Color
-}
-
-// GetNoColor returns the value of the `token` flag.
-func (a ParsedArgsFromRust) GetNoColor() bool {
-	return a.NoColor
-}
-
-// GetLogin returns the value of the `login` flag.
-func (a ParsedArgsFromRust) GetLogin() (string, error) {
-	return a.Login, nil
-}
-
-// GetAPI returns the value of the `api` flag.
-func (a ParsedArgsFromRust) GetAPI() (string, error) {
-	return a.API, nil
-}
-
-// GetTeam returns the value of the `team` flag.
-func (a ParsedArgsFromRust) GetTeam() (string, error) {
-	return a.Team, nil
-}
-
-// GetToken returns the value of the `token` flag.
-func (a ParsedArgsFromRust) GetToken() (string, error) {
-	return a.Token, nil
-}
-
-// GetCwd returns the value of the `cwd` flag.
-func (a ParsedArgsFromRust) GetCwd() (string, error) {
-	return a.CWD, nil
+// ExecutionState is the entire state of a turbo execution that is passed from the Rust shim.
+type ExecutionState struct {
+	APIClientConfig APIClientConfig    `json:"remote_config"`
+	CLIArgs         ParsedArgsFromRust `json:"cli_args"`
 }
 
-// GetRemoteCacheTimeout returns the value of the `remote-cache-timeout` flag.
-func (a ParsedArgsFromRust) GetRemoteCacheTimeout() (uint64, error) {
-	if a.RemoteCacheTimeout != 0 {
-		return a.RemoteCacheTimeout, nil
-	}
-	return 0, fmt.Errorf("no remote cache timeout provided")
+// APIClientConfig holds the authentication and endpoint details for the API client
+type APIClientConfig struct {
+	Token        string `json:"token"`
+	TeamID       string `json:"team_id"`
+	TeamSlug     string `json:"team_slug"`
+	APIURL       string `json:"api_url"`
+	UsePreflight bool   `json:"use_preflight"`
+	Timeout      uint64 `json:"timeout"`
```

Comment:
We can do this on the rust side, but it would be good to document where these values can come from  (flag, xdg config, `./.turbo/config.json`, env var, etc). I always forget that `teamId` and `token` are in separate places

Reply:
yeah but there are 2 different config files. hard to remember which one has what

Reply:
Gotcha. Added a comment to make it clearer